packages/blade/src/components/FileUpload/_decisions/decisions.md

```diff
@@ -0,0 +1,398 @@
+- [Examples](#examples)
+  - [Uncontrolled Usage](#uncontrolled-usage)
+    - [Single File selection](#single-file-selection)
+    - [Multiple File selection:](#multiple-file-selection)
+  - [Controlled Usage](#controlled-usage)
+    - [Direct Upload with Progress Display](#direct-upload-with-progress-display)
+- [Accessibility](#accessibility)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade-DSL?type=design&node-id=78150%3A2625&mode=design&t=bKQvPInRmAoeAAFV-1) to all variants of the `FileUpload` component.
+
+## Anatomy
+
+<img width="100%" src="./file-upload-anatomy.png" alt="Single File Upload">
+<img width="100%" src="./file-upload-item-anatomy.png" alt="Multiple Files Upload">
+
+## `FileUpload` Props
+
+```ts
+// Check the File type from MDN for more details: https://developer.mozilla.org/en-US/docs/Web/API/File
+interface BladeFile extends File {
+  /**
+   * The unique identifier of the file.
+   */
+  id: string;
+  /**
+   * The file's upload status.
+   */
+  status?: 'uploading' | 'success' | 'error';
+  /**
+   * The percentage of file upload completion.
+   */
+  percent?: number;
+  /**
+   * Text indicating an error state
+   */
+  errorText?: string;
+}
+
+type BladeFileList = BladeFile[];
+
+type FileUploadCommonProps = {
+  /**
+   * Position of the label relative to the file upload area. Desktop only prop. Default value on mobile will be 'top'
+   *
+   * @default 'top'
+   */
+  labelPosition?: 'top' | 'left';
+  /**
+   * Defines the selection behavior of the FileUpload component
+   */
+  selectionType?: 'single' | 'multiple';
+  /**
+   * File types that can be accepted. See [input's accept attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#accept)
+   *
+   * Usage: accept=".jpg, .png, .pdf", accept="image/*", accept="image/png, image/jpeg, application/pdf"
+   */
+  accept?: string;
+  /**
+   * Disables or enables the FileUpload component
+   */
+  isDisabled?: boolean;
+  /**
+   * Sets the required state of the file input
+   *
+   * @default false
+   */
+  isRequired?: boolean;
+  /**
+   * Renders a necessity indicator after the label. If `isRequired` is `true`, it defaults to `'required'`
+   */
+  necessityIndicator?: 'required' | 'optional' | 'none';
+  /**
+   * The name of the file upload input, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name)
+   */
+  name?: string;
+  /**
+   * Default list of files that have been uploaded, useful when the component is uncontrolled
+   */
+  defaultFileList?: BladeFileList;
+  /**
+   * List of files that have been selected/uploaded, useful when the component is controlled
+   */
+  fileList?: BladeFileList;
+  /**
+   * Limit the number of files that can be uploaded
+   */
+  maxCount?: number;
+  /**
+   * Limit the size of the uploaded files (in bytes)
+   */
+  maxSize?: number;
```

Comment:
For multiple files does it limit total size or per file?

Reply:
This is per file, I'll update the description to make it clear.